### PR TITLE
Add flake8 checks

### PR DESCRIPTION
**Pull request by vojtechsokol**
Add flake8 check to the `lint` Makefile target and make the linter happy by fixing most of the issues.
The only remaining issues are:
- F811: redefinition of unused 'current_actor_context'                                                                                                                                                                       
- F811: redefinition of unused 'current_actor_libraries'                                                                                                                                                                                   
- F401: 'leapp.snactor.fixture.current_actor_context' imported but unused                                                                                                                                                                  
- F401: 'leapp.snactor.fixture.current_actor_libraries' imported but unused                                                                                                                                                                

All those issues are specific to actors' tests and are ignored via flake8 config in `setup.cfg` file.

### CODE REVIEW

File: repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py [hunk 1]
```diff
@@ -2,6 +2,8 @@
 from leapp.libraries.stdlib import api, run
 
 # Check if the chrony config file was not modified since installation
+
+
```
Comment (jmikovic) [inline repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py]: add 1 blank line before the comment instead of 2 between comment and function, so it's clear what the comment refers to

File: repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py [hunk 2]
```diff
@@ -11,18 +13,21 @@ def is_config_default():
         return True
 
 # Report potential issues in chrony configuration
+
+
```
Comment (jmikovic) [inline repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py]: same as above

File: repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py [hunk 3]
```diff
@@ -2,6 +2,8 @@
 from leapp.libraries.stdlib import api, run
 
 # Check if the chrony config file was not modified since installation
+
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py]: Wrong change IMHO

File: repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py [hunk 4]
```diff
@@ -2,6 +2,8 @@
 from leapp.libraries.stdlib import api, run
 
 # Check if the chrony config file was not modified since installation
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py]:
```suggestion

```

File: repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py [hunk 5]
```diff
@@ -2,6 +2,8 @@
...
+
+
 def is_config_default():
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py]:
```suggestion
def is_config_default():
    """ Check if the chrony config file was not modified since installation """
```

File: repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py [hunk 6]
```diff
@@ -11,18 +13,21 @@ def is_config_default():
         return True
 
 # Report potential issues in chrony configuration
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py]:
```suggestion
```

File: repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py [hunk 7]
```diff
@@ -11,18 +13,21 @@ def is_config_default():
...
+
+
 def check_chrony(chrony_installed):
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/checkchrony/libraries/library.py]:
```suggestion
def check_chrony(chrony_installed):
    """ Report potential issues in chrony configuration """
```

File: repos/system_upgrade/el7toel8/actors/peseventsscanner/tests/unit_test.py [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import os
+import os  # noqa: F401
```
Comment (jmikovic) [inline repos/system_upgrade/el7toel8/actors/peseventsscanner/tests/unit_test.py]: why is this import needed?

File: repos/system_upgrade/el7toel8/actors/pythoninformuser/actor.py [hunk 9]
```diff
@@ -12,7 +12,7 @@ class PythonInformUser(Actor):
     tags = (ChecksPhaseTag,)
 
     def process(self):
-        url = "https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/configuring_basic_system_settings/#using-python3" # noqa E501
+        url = "https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/configuring_basic_system_settings/#using-python3"  # noqa E501
```
Comment (jmikovic) [inline repos/system_upgrade/el7toel8/actors/pythoninformuser/actor.py]:
```suggestion
        url = "https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/configuring_basic_system_settings/#using-python3"  # noqa: E501
```

File: repos/system_upgrade/el7toel8/actors/peseventsscanner/tests/unit_test.py (hunk 8 above)
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/peseventsscanner/tests/unit_test.py]: test_pes_data_not_found should change instead of:
```python
monkeypatch.setattr('os.path.isfile', file_not_exists)
```
to
```python
monkeypatch.setattr(os.path, 'isfile', file_not_exists)
```

But I might be mistaken about this one, also since there's no other use, I'd even go with import os.path to make clear that os.path is what is wanted.

File: repos/system_upgrade/el7toel8/actors/prepareupgradetransaction/libraries/xinitramgen.py [hunk 10]
```diff
@@ -97,12 +97,14 @@ def generate_initram_disk(userspace_dir):
         '/bin/sh', '/generate-initram.sh'
     ])
 
+
 def remove_userspace(userspace_dir):
     try:
         run(['rm', '-rf', userspace_dir])
     except Exception:
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/prepareupgradetransaction/libraries/xinitramgen.py]: except (OSError, CalledProcessError)

File: repos/system_upgrade/el7toel8/actors/peseventsscanner/tests/unit_test.py (hunk 8 above)
Comment (vinzenz) [reply repos/system_upgrade/el7toel8/actors/peseventsscanner/tests/unit_test.py]: @jmikovic see my comment

File: repos/system_upgrade/el7toel8/actors/redhatsignedrpmcheck/tests/test_redhatsignedrpmcheck.py [hunk 11]
```diff
@@ -30,7 +30,7 @@ def __call__(self, **report_fields):
 
 
 def test_skip_check(monkeypatch):
-    monkeypatch.setattr(os, "getenv", lambda(_): True)
+    monkeypatch.setattr(os, "getenv", lambda(_): True)  # noqa: E999
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/redhatsignedrpmcheck/tests/test_redhatsignedrpmcheck.py]: actually using _ is forbidden in leapp, due to localization strings.
```suggestion
    monkeypatch.setattr(os, "getenv", lambda(_unused): True)  # noqa: E999
```

File: repos/system_upgrade/el7toel8/actors/removeupgradebootentry/tests/unit_test.py [hunk 12]
```diff
@@ -1,7 +1,7 @@
 import pytest
 
 from leapp.exceptions import StopActorExecutionError
-from leapp.libraries import stdlib
+from leapp.libraries import stdlib  # noqa: F401
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/removeupgradebootentry/tests/unit_test.py]:
```suggestion
from leapp.libraries.stdlib import api
```

and later in the tests there should be `setattr(api, 'consume', ....` and `setattr(api, 'produce', ...`

File: repos/system_upgrade/el7toel8/actors/redhatsignedrpmcheck/tests/test_redhatsignedrpmcheck.py (hunk 11 above)
Comment (vojtechsokol) [reply repos/system_upgrade/el7toel8/actors/redhatsignedrpmcheck/tests/test_redhatsignedrpmcheck.py]: Thanks. I completely forgot about that. Is it mentioned somewhere? Guidelines, conding standards, etc.?

Comment (vinzenz) [reply repos/system_upgrade/el7toel8/actors/redhatsignedrpmcheck/tests/test_redhatsignedrpmcheck.py]: Not sure, it might have been forgotten to add

**Comment by vojtechsokol** [comment]
@fernflower Done

**Comment by vojtechsokol** [comment]
To whomever who merges this PR: please squash and merge.

File: repos/system_upgrade/el7toel8/actors/vsftpdconfigread/libraries/library.py [hunk 13]
```diff
@@ -3,9 +3,9 @@
 
 from leapp.libraries.actor import config_parser
 from leapp.libraries.common.vsftpdutils import get_config_contents, get_default_config_hash, \
```
Comment (fernflower) [inline repos/system_upgrade/el7toel8/actors/vsftpdconfigread/libraries/library.py]: Imho L5-L88 are candidates for module level import

File: setup.cfg [hunk 14]
```diff
@@ -0,0 +1,8 @@
+[flake8]
+max-line-length = 119
+per-file-ignores =
+    # redefinition of unused 'current_actor_context'
```
Comment (fernflower) [inline setup.cfg]: Imho we should not keep commented code unless it's a quickfix.

File: .pylintrc [hunk 15]
```diff
@@ -7,12 +7,12 @@ disable=
  no-member,
  no-name-in-module,
  raising-bad-type,
- undefined-variable,
+ #undefined-variable,
```
Comment (fernflower) [inline .pylintrc]: I'd rather not commit commented out code

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py [hunk 16]
```diff
@@ -50,15 +50,15 @@ class ReportsPhase(Phase):
 
     # The following two phases have been removed but are kept here for reference or in case they are needed again
 
-    #class AttachPackageReposPhase(Phase):
+    # class AttachPackageReposPhase(Phase):
```
Comment (fernflower) [inline repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: Open question: Are we going need these 2 classes in the nearest future? 
If not I'd suggest to remove them, git knows what we did last summer.

File: repos/system_upgrade/el7toel8/actors/storagescanner/tests/unit_test.py [hunk 17]
```diff
@@ -15,7 +15,7 @@ def is_file_readable_mocked(path):
         PartitionEntry(major='253', minor='1', blocks='1048576', name='dm-1')]
     assert expected == library._get_partitions_info('tests/files/partitions')
 
-    monkeypatch.setattr('library._is_file_readable', is_file_readable_mocked)
+    monkeypatch.setattr(library, '_is_file_readable', is_file_readable_mocked)
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/actors/storagescanner/tests/unit_test.py]: I think you have to rebase

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py (hunk 16 above)
Comment (vinzenz) [reply repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: We kept them here for reference, even if we can ask git about what we did last summer. Any bigger refactoring makes git scream and run around like a headless chicken with arms waving all around forcing us to ask them multiple times...

Anyway I know where you are going, but we might want to remove it. Just not with this PR.

File: setup.cfg (hunk 14 above)
Comment (vojtechsokol) [reply setup.cfg]: Well those lines were just a hint why we ignore F811 (redefinition) and F401 (unused imports).

File: repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py (hunk 16 above)
Comment (vojtechsokol) [reply repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py]: Closing this in favor of separate PR.

File: repos/system_upgrade/el7toel8/actors/storagescanner/tests/unit_test.py (hunk 17 above)
Comment (vojtechsokol) [reply repos/system_upgrade/el7toel8/actors/storagescanner/tests/unit_test.py]: Done

**Comment by vinzenz** [comment]
Meh. I merged the change which causes this to fail, I didn’t realize the spaces

Sorry
